docs/features/nullable-reference-types.md

```diff
@@ -0,0 +1,119 @@
+Nullable Reference Types
+=========
+References to reference types may be nullable, non-nullable, or null-oblivious. (The document uses the shorthand `?`, `!`, and `~`.)
+
+## Annotations
+In source, nullable reference types are annotated with `?`.
+```c#
+string? OptString; // may be null
+List<object?>? OptListOptElements; // list may be null, elements may be null
+```
+In metadata, nullable reference types are annotated with a `[Nullable]` attribute.
+```c#
+namespace System.Runtime.CompilerServices
+{
+    public sealed class NullableAttribute : Attribute
```

Comment:
Let's add the valid targets

Reply:
Why would you put this attribute on a class or module?

---
In reply to: [194861205](https://github.com/dotnet/roslyn/pull/27721#discussion_r194861205) [](ancestors = 194861205)

Reply:
Removed module. Class is used as the attribute target for base type nullability.

---
In reply to: [194872780](https://github.com/dotnet/roslyn/pull/27721#discussion_r194872780) [](ancestors = 194872780,194861205)